superset-frontend/src/components/Select/common.tsx

```diff
@@ -0,0 +1,363 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { styled, t } from '@superset-ui/core';
+import { Spin } from 'antd';
+import Icons from 'src/components/Icons';
+import AntdSelect, {
+  SelectProps as AntdSelectProps,
+  SelectValue as AntdSelectValue,
+  LabeledValue as AntdLabeledValue,
+} from 'antd/lib/select';
+import { rankedSearchCompare } from 'src/utils/rankedSearchCompare';
+import React, {
+  forwardRef,
+  ReactElement,
+  ReactNode,
+  RefObject,
+  JSXElementConstructor,
+} from 'react';
+
+const { Option } = AntdSelect;
+
+export type AntdSelectAllProps = AntdSelectProps<AntdSelectValue>;
+
+export type PickedSelectProps = Pick<
+  AntdSelectAllProps,
+  | 'allowClear'
+  | 'autoFocus'
+  | 'disabled'
+  | 'filterOption'
+  | 'filterSort'
+  | 'loading'
+  | 'labelInValue'
+  | 'maxTagCount'
+  | 'notFoundContent'
+  | 'onChange'
+  | 'onClear'
+  | 'onDeselect'
+  | 'onSelect'
+  | 'onFocus'
+  | 'onBlur'
+  | 'onPopupScroll'
+  | 'onSearch'
+  | 'onDropdownVisibleChange'
+  | 'placeholder'
+  | 'showArrow'
+  | 'showSearch'
+  | 'tokenSeparators'
+  | 'value'
+  | 'getPopupContainer'
+  | 'menuItemSelectedIcon'
+>;
+
+export type SelectOptionsType = Exclude<
+  AntdSelectAllProps['options'],
+  undefined
+>;
+
+export type SelectOptionsTypePage = {
+  data: SelectOptionsType;
+  totalCount: number;
+};
+
+export type SelectOptionsPagePromise = (
+  search: string,
+  page: number,
+  pageSize: number,
+) => Promise<SelectOptionsTypePage>;
+
+export const StyledContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+`;
+
+export const StyledSelect = styled(AntdSelect)`
+  ${({ theme }) => `
+    && .ant-select-selector {
+      border-radius: ${theme.gridUnit}px;
+    }
+    // Open the dropdown when clicking on the suffix
+    // This is fixed in version 4.16
+    .ant-select-arrow .anticon:not(.ant-select-suffix) {
+      pointer-events: none;
+    }
+  `}
+`;
+
+export const StyledStopOutlined = styled(Icons.StopOutlined)`
+  vertical-align: 0;
+`;
+
+export const StyledCheckOutlined = styled(Icons.CheckOutlined)`
+  vertical-align: 0;
+`;
+
+export const StyledError = styled.div`
```

Comment:
Can we move error components into the async file since the error only exists for async operations?